src/app/services/user-deposit.service.ts

```diff
@@ -311,4 +369,84 @@ export class UserDepositService {
             shareReplay({ bufferSize: 1, refCount: true })
         );
     }
+
+    private getWithdrawPlanObservable(
+        userDepositContract$: Observable<Contract>
+    ): Observable<WithdrawPlan> {
+        return this.withdrawPlanSubject.pipe(
+            switchMapTo(
+                zip(this.raidenService.raidenAddress$, userDepositContract$)
+            ),
+            switchMap(([raidenAddress, userDepositContract]) =>
+                from<Promise<{ amount: string; withdraw_block: string }>>(
+                    userDepositContract.methods
+                        .withdraw_plans(raidenAddress)
+                        .call()
+                )
+            ),
+            map(
+                ({ amount, withdraw_block }) =>
+                    ({
+                        amount: new BigNumber(amount),
+                        withdrawBlock: Number(withdraw_block),
+                    } as WithdrawPlan)
+            ),
+            retryWhen((errors) =>
+                errors.pipe(switchMapTo(this.raidenService.rpcConnected$))
+            ),
+            shareReplay(1)
```

Comment:
Just a headsup: `share*` operators can be a little dangerous sometimes, because they only stay hot while they have at least one subscriber; since this seems to be a readonly observable, this would be just an optimization, but in places where you use it as source to actions, it´s important to keep this fact in mind. Good job here, nice usage of the operators